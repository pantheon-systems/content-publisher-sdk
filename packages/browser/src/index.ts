import { PantheonClient } from "@pantheon-systems/pcc-sdk-core";
import PCCArticle from "./elements/pcc-article";

type Args = {
  siteId?: string;
  apiKey?: string;
  pccHost?: string;
};

const { siteId, apiKey, pccHost } = document.currentScript?.dataset as Args;

// Validate required arguments
if (!siteId) {
  throw new Error(
    "Missing Pantheon Content Cloud site ID. Provide it as a data-site-id attribute on the script tag.",
  );
}
if (!apiKey) {
  throw new Error(
    "Missing Pantheon Content Cloud API Key. Provide it as a data-api-key attribute on the script tag.",
  );
}

// Initialize the PantheonClient
const pantheonClient = new PantheonClient({
  siteId,
  apiKey,
  pccHost,
});

// Expose the PantheonClient on the window object
if (!window?.__PANTHEON_CLIENT) window.__PANTHEON_CLIENT = pantheonClient;

// Register the custom element
customElements.define("pcc-article", PCCArticle);
