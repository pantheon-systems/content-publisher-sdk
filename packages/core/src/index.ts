export * from "./core/pantheon-client";
export * from "./core/pantheon-api";

export * from "./lib/apollo-client";
export * from "./lib/gql";
export * as GQL from "./lib/gql";
export * from "./lib/jwt";

export * from "./helpers";

export * from "./types";
