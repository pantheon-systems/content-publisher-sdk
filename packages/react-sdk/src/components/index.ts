export {
  default as ArticleRenderer,
  ServersideSmartComponentMap,
  SmartComponentMap,
} from "./ArticleRenderer";
