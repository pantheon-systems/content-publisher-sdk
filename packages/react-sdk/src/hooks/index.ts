export { useArticle } from "./use-article";
export { useArticles } from "./use-articles";
export { usePagination } from "./use-pagination";

export { useQuery } from "@apollo/client/react/hooks/useQuery.js";
export { useMutation } from "@apollo/client/react/hooks/useMutation.js";
export { useSubscription } from "@apollo/client/react/hooks/useSubscription.js";
