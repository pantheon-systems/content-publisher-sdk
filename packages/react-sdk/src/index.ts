export type {
  PantheonClientConfig,
  PantheonAPIOptions,
} from "@pantheon-systems/pcc-sdk-core";
export { PantheonClient } from "@pantheon-systems/pcc-sdk-core";

export {
  getArticles,
  getPaginatedArticles,
  getArticle,
  getAllTags,
  getArticleBySlugOrId,
  PCCConvenienceFunctions,
} from "@pantheon-systems/pcc-sdk-core";
export * from "@pantheon-systems/pcc-sdk-core/types";

export * from "./core/pantheon-context";

export { NextPantheonAPI as PantheonAPI } from "./core/pantheon-api";

// Would be nice to have this as a separate entry point but
// context is defined at module scope so can't be shared between
// entry points.
// https://github.com/facebook/react/issues/19541
export * from "./hooks";
