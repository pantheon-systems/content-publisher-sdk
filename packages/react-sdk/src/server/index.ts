export type {
  PantheonClientConfig,
  PantheonAPIOptions,
} from "@pantheon-systems/pcc-sdk-core";
export { PantheonClient } from "@pantheon-systems/pcc-sdk-core";

export {
  getArticles,
  getPaginatedArticles,
  getRecommendedArticles,
  getArticle,
  getAllTags,
  getArticleBySlugOrId,
  PCCConvenienceFunctions,
  updateConfig,
} from "@pantheon-systems/pcc-sdk-core";
export * from "@pantheon-systems/pcc-sdk-core/types";

export * from "../core/pantheon-context";

export { NextPantheonAPI as PantheonAPI } from "../core/pantheon-api";
