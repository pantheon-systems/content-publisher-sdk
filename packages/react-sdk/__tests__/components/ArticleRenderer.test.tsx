/**
 * @jest-environment jsdom
 */

import { Article } from "@pantheon-systems/pcc-sdk-core/types";
import { render } from "@testing-library/react";
import { ArticleRenderer, getArticleTitle } from "../../src/components";
import articleWithImageMarkdown from "../data/article-with-image-markdown.json";
import articleWithImageTree from "../data/article-with-image-tree.json";
import article from "../data/article.json";

describe("<ArticleRenderer />", () => {
  it("should render a post's content", () => {
    const { container } = render(
      <ArticleRenderer
        article={article as Article}
        __experimentalFlags={{ useUnintrusiveTitleRendering: true }}
      />,
    );
    const title = getArticleTitle(article as Article);
    expect(title).toEqual("Test 1");
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should replace the CDN URL with the override (markdown)", () => {
    const { container } = render(
      <ArticleRenderer
        article={articleWithImageMarkdown as Article}
        __experimentalFlags={{ cdnURLOverride: "cdn.example.com" }}
      />,
    );
    expect(container.innerHTML).toMatch("https://cdn.example.com");
    expect(container.innerHTML).not.toMatch("https://cdn.staging.content");
    expect(container.firstChild).toMatchSnapshot();
  });

  it.skip("should replace the CDN URL with the override (tree)", () => {
    const { container } = render(
      <ArticleRenderer
        article={articleWithImageTree as Article}
        __experimentalFlags={{ cdnURLOverride: "cdn.example.com" }}
      />,
    );
    expect(container.innerHTML.includes("https://cdn.example.com")).toBe(true);
    expect(container.innerHTML.includes("https://cdn.staging.content")).toBe(
      false,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
