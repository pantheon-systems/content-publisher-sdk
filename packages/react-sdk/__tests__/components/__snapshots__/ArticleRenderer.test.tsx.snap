// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleRenderer /> should render a post's content 1`] = `
<div>
  <div>
    <h1>
      <span>
        This is a title, there are many like it, but this one is mine
      </span>
    </h1>
    <h1>
      <span>
        This is h1
      </span>
    </h1>
    <h2>
      <span>
        This is h2
      </span>
    </h2>
    <h3>
      <span>
        This is h3
      </span>
    </h3>
    <h4>
      <span>
        This is h4
      </span>
    </h4>
    <h5>
      <span>
        This is h5
      </span>
    </h5>
    <h6>
      <span
        style="font-style: italic;"
      >
        This is h6
      </span>
    </h6>
    <div />
    <div>
      <span
        style="font-weight: bold;"
      >
        This is bold
      </span>
    </div>
    <div>
      <span
        style="font-style: italic;"
      >
        This is italic
      </span>
    </div>
    <div>
      <span
        style="text-decoration: underline;"
      >
        This is underlined
      </span>
    </div>
    <div />
    <div>
      <span
        style="vertical-align: super;"
      >
        aaa
      </span>
      <span
        style="font-weight: bold; font-style: italic; vertical-align: super; text-decoration: underline;"
      >
        This is bold, underlined, italic, strikethrough and supertext
      </span>
    </div>
    <div>
      <span
        style="vertical-align: sub;"
      >
        eee
      </span>
      <span
        style="font-weight: bold; font-style: italic; vertical-align: sub; text-decoration: underline;"
      >
        This is bold, underlined, italic, strikethrough and subscript
      </span>
    </div>
    <div />
    <div>
      <span
        style="vertical-align: sub;"
      >
        horizontal line below:
      </span>
    </div>
    <hr />
    <div />
    <div />
    <div>
      <span>
        x
      </span>
      <span
        style="vertical-align: super;"
      >
        3
      </span>
    </div>
    <div />
    <div>
      <span
        style="vertical-align: super;"
      >
        H
      </span>
      <span
        style="vertical-align: sub;"
      >
        2
      </span>
      <span
        style="vertical-align: super;"
      >
        O x
      </span>
      <span
        style="font-weight: bold; vertical-align: super;"
      >
        o
      </span>
    </div>
    <div />
    <div>
      <span
        style="vertical-align: sub;"
      >
        Unordered “bullet point” list
      </span>
    </div>
    <ul>
      <li>
        <span
          style="vertical-align: sub;"
        >
          Apple
        </span>
      </li>
      <li>
        <span
          style="vertical-align: sub;"
        >
          Pineapple
        </span>
      </li>
      <li>
        <span
          style="vertical-align: sub;"
        >
          Pen 
        </span>
        <span
          style="font-weight: bold; vertical-align: sub;"
        >
          pineapple
        </span>
        <span
          style="font-style: italic; vertical-align: sub;"
        >
          apple
        </span>
        <span
          style="vertical-align: sub; text-decoration: underline;"
        >
          pen
        </span>
      </li>
    </ul>
    <div />
    <div>
      <span
        style="vertical-align: sub;"
      >
        Unordered list with different symbols
      </span>
    </div>
    <ul>
      <li>
        <span
          style="vertical-align: sub;"
        >
          Duck
        </span>
      </li>
      <li>
        <span
          style="vertical-align: sub;"
        >
          Duck2
        </span>
      </li>
      <li>
        <span
          style="vertical-align: sub;"
        >
          Goose
        </span>
      </li>
    </ul>
    <div />
    <div>
      <span
        style="vertical-align: sub;"
      >
        Todo list with checkmarks
      </span>
    </div>
    <ul>
      <li>
        <span
          style="vertical-align: sub;"
        >
          Veni
        </span>
      </li>
      <li>
        <span
          style="vertical-align: sub;"
        >
          Vidi
        </span>
      </li>
      <li>
        <span
          style="vertical-align: sub;"
        >
          Vici
        </span>
      </li>
    </ul>
    <div />
    <div>
      <span
        style="vertical-align: sub;"
      >
        Ordered “numbered” list
      </span>
    </div>
    <ol>
      <li>
        <span
          style="vertical-align: sub;"
        >
          Waking in the morning
        </span>
      </li>
      <li>
        <span
          style="vertical-align: sub;"
        >
          Gotta go downstairs
        </span>
      </li>
      <li>
        <span
          style="vertical-align: sub;"
        >
          Gotta have cereal
        </span>
      </li>
      <li>
        <span
          style="vertical-align: sub;"
        >
          I see my friends
        </span>
      </li>
      <li>
        <span
          style="vertical-align: sub;"
        >
          Gotta get down on Friday
        </span>
      </li>
    </ol>
    <div />
    <div>
      <img />
    </div>
    <div />
    <div>
      <span
        style="font-weight: bold; vertical-align: sub;"
      >
        A wild table appears
      </span>
    </div>
    <table
      class="text-black"
    >
      <tbody>
        <tr>
          <td>
            <div>
              <span
                style="vertical-align: sub;"
              >
                Merged cols 1 & 2
              </span>
            </div>
          </td>
          <td>
            <div>
              <span
                style="vertical-align: sub;"
              >
                Col 3
              </span>
            </div>
          </td>
          <td>
            <div>
              <span
                style="vertical-align: sub;"
              >
                Col 4
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <span
                style="vertical-align: sub;"
              >
                A
              </span>
            </div>
          </td>
          <td>
            <div>
              <span
                style="vertical-align: sub;"
              >
                B
              </span>
            </div>
          </td>
          <td>
            <div>
              <span
                style="vertical-align: sub;"
              >
                C
              </span>
            </div>
          </td>
          <td>
            <div>
              <span
                style="vertical-align: sub;"
              >
                D
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <td>
            <div>
              <span
                style="vertical-align: sub;"
              >
                ა
              </span>
            </div>
          </td>
          <td>
            <div>
              <span
                style="vertical-align: sub;"
              >
                ბ
              </span>
            </div>
          </td>
          <td>
            <div>
              <span
                style="vertical-align: sub;"
              >
                ს
              </span>
            </div>
          </td>
          <td>
            <div>
              <span
                style="vertical-align: sub;"
              >
                დ
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <div />
    <div>
      <span
        style="vertical-align: sub;"
      >
        Some more paragraphs after everything else.
      </span>
    </div>
    <div />
    <div />
  </div>
</div>
`;

exports[`<ArticleRenderer /> should replace the CDN URL with the override (markdown) 1`] = `
<div>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque vitae leo nec massa semper venenatis eu at felis. Nam consequat accumsan dictum. Suspendisse eu metus in justo eleifend consequat id quis nunc. Etiam pharetra, tortor id malesuada efficitur, ligula sem posuere erat, vitae placerat ipsum felis non leo. Praesent lobortis, mauris vel posuere bibendum, leo diam vehicula lacus, a interdum odio massa et elit. Suspendisse ac lorem imperdiet, suscipit quam eu, maximus elit. Quisque nec metus eu nibh iaculis dignissim. Cras ultricies pulvinar erat eget luctus. In id ipsum non diam sagittis ullamcorper eget nec arcu. Ut laoreet accumsan ex non porta. Donec vitae mi porttitor, dictum lacus sit amet, dapibus ipsum. Praesent bibendum lacus nibh, sed egestas enim facilisis vitae!
    <img
      alt=""
      src="https://cdn.example.com/loAWY0YB0HTHexSzw3Z1/djHVTYbPaCby44H5CxhH"
      title="just another dog"
    />
  </p>
  

  <p>
    Integer pretium bibendum ipsum, vehicula placerat orci finibus id. Proin orci mauris, ornare commodo congue sit amet, dignissim non eros. Nullam risus nibh, dictum et enim vel, vestibulum maximus velit. Proin eu porta risus. Aliquam nec lacus et lacus commodo elementum. Mauris quis massa ac urna iaculis hendrerit. Nulla eget ullamcorper lorem, in rutrum sapien. Vivamus finibus placerat massa a mollis.
  </p>
  

  <p>
    Cras vitae ultricies odio. Vestibulum iaculis interdum feugiat. Donec iaculis finibus lorem. Quisque lacinia volutpat magna, in dapibus lectus gravida id. Cras iaculis justo et diam elementum, nec rhoncus dolor efficitur. Sed lacus libero, lobortis in mauris ut, imperdiet suscipit quam. Sed eu leo vitae sapien vestibulum viverra.
  </p>
  

  <p>
    Morbi varius, velit volutpat auctor malesuada, nisi mi rhoncus urna, at tempus risus libero ac turpis. Nullam sit amet porta leo. Nam dapibus maximus ante, vitae fringilla ipsum lacinia posuere. Donec efficitur, odio ac blandit convallis, purus arcu posuere risus, maximus scelerisque quam augue id ante. Etiam ultrices neque in purus rhoncus laoreet. Nunc bibendum tincidunt felis, quis molestie libero dictum scelerisque. Sed urna dolor, elementum eget auctor tincidunt, lacinia in diam. Duis a libero tristique, luctus lacus aliquet, fermentum odio. Integer lacinia nunc metus, ac egestas nibh mattis a. Fusce sodales mollis odio non pulvinar. Nulla vitae dignissim nibh, ac finibus neque.
  </p>
  

  <p>
    Mauris mollis, lacus eu sodales tincidunt, ipsum ex pharetra libero, eu posuere ante lectus nec velit. Sed commodo ultrices diam eget pulvinar. Vestibulum volutpat sapien non nisl porttitor, sed euismod magna sollicitudin. Suspendisse potenti. Sed posuere sapien id magna euismod tincidunt. Phasellus in cursus diam, eget vestibulum ipsum. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Curabitur tempus mollis tempus. Phasellus rhoncus, turpis sed mollis rutrum, tellus mauris hendrerit nulla, a congue dolor mauris in leo. Sed feugiat posuere neque, eget cursus est luctus at. Vestibulum accumsan convallis ante in auctor. Aliquam ac orci sem. Nunc auctor mauris sed felis tristique, nec malesuada ante porta.
  </p>
  

  <p>
    Sed et ligula nec nisl convallis volutpat. Curabitur semper massa ac mollis feugiat. Vestibulum eget orci ac arcu tempus dapibus. Nunc augue lorem, ultrices in velit non, porta cursus ex. Aenean eget nunc sit amet quam vulputate porttitor vitae sed ex. Morbi porttitor libero nec hendrerit vehicula. Ut maximus odio ac odio blandit pharetra. Etiam quis condimentum est. Morbi dui nisi, ornare at dignissim vel, ultrices rhoncus nisl. Sed at euismod enim. Quisque efficitur orci a velit pretium scelerisque. Nulla eleifend feugiat porta. Etiam eu rhoncus lectus.
    <img
      alt=""
      src="https://cdn.example.com/loAWY0YB0HTHexSzw3Z1/ZWJn6ilXOYs9oS9iDgMH"
      title="This is a title"
    />
  </p>
  

  <p>
    THIS PUPPY BREAKS UP TEXT Etiam maximus mauris eu felis interdum, in bibendum orci sodales. In posuere elit quam. Nulla dictum, sapien ut sagittis consequat, nisl elit suscipit lectus, a blandit risus nunc eget nunc. Donec augue ante, eleifend non risus at, dignissim gravida ipsum. Vivamus tristique tempor dui, at condimentum dui tristique nec. Suspendisse potenti. Nulla facilisi. Quisque orci risus, iaculis vel vehicula eu, cursus sit amet justo. Cras sed erat a est tristique elementum.
  </p>
  

  <p>
    Morbi justo lorem, ultricies ut volutpat aliquam, blandit eu erat. Duis accumsan egestas convallis. Morbi a finibus lectus. Vivamus vel ultrices felis. Donec faucibus placerat mi, non pellentesque est lacinia non. Ut vitae metus leo. Mauris volutpat malesuada lorem, et auctor metus sodales ultrices. Nulla rhoncus augue pulvinar consectetur bibendum. Nunc ac urna vel mauris venenatis euismod in et massa.
  </p>
  

  <p>
    Proin sed laoreet augue. Aliquam malesuada vel ex ut vestibulum. Mauris at libero vestibulum, vehicula mi nec, ullamcorper magna. Vivamus bibendum ipsum imperdiet placerat pharetra. Curabitur fermentum id est a tristique. Integer at semper erat. Phasellus volutpat dolor in ipsum hendrerit, eu maximus libero mattis. Nunc aliquam, tortor at congue vehicula, mi purus mollis metus, et convallis orci quam at elit.
    <img
      alt=""
      src="https://cdn.example.com/loAWY0YB0HTHexSzw3Z1/RoJOqZdOWPaxWCVnH9sD"
    />
  </p>
  

  <p>
    Etiam non eleifend nisl. Morbi eu libero a erat volutpat dictum eu sodales urna. Ut venenatis rutrum dolor, a euismod nibh egestas a. Nullam pretium congue tellus, eu tempor est consectetur et. Etiam dapibus consequat volutpat. Nunc sed bibendum ipsum, a egestas risus. Nulla id dolor nibh. Nulla tempor diam eget sapien elementum faucibus. Sed elementum, mi ut maximus facilisis, ex enim fringilla sem, at fermentum odio mi id diam. Pellentesque vitae commodo felis. Vestibulum eget diam pellentesque, porttitor ante a, ultricies lacus.
    <img
      alt=""
      src="https://cdn.example.com/loAWY0YB0HTHexSzw3Z1/RoJOqZdOWPaxWCVnH9sD"
    />
  </p>
  

  <p>
    <img
      alt=""
      src="https://cdn.example.com/loAWY0YB0HTHexSzw3Z1/RoJOqZdOWPaxWCVnH9sD"
    />
  </p>
  

  <p>
    <img
      alt=""
      src="https://cdn.example.com/loAWY0YB0HTHexSzw3Z1/RoJOqZdOWPaxWCVnH9sD"
      title="inline puppies"
    />
  </p>
</div>
`;

exports[`<ArticleRenderer /> should replace the CDN URL with the override (tree) 1`] = `
<div>
  <div>
    <p
      className="normal-text"
      style={
        {
          "margin": "0",
          "marginBottom": "11PT",
        }
      }
    >
      <span
        className=""
      >
        <span
          className=""
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque vitae leo nec massa semper venenatis eu at felis. Nam consequat accumsan dictum. Suspendisse eu metus in justo eleifend consequat id quis nunc. Etiam pharetra, tortor id malesuada efficitur, ligula sem posuere erat, vitae placerat ipsum felis non leo. Praesent lobortis, mauris vel posuere bibendum, leo diam vehicula lacus, a interdum odio massa et elit. Suspendisse ac lorem imperdiet, suscipit quam eu, maximus elit. Quisque nec metus eu nibh iaculis dignissim. Cras ultricies pulvinar erat eget luctus. In id ipsum non diam sagittis ullamcorper eget nec arcu. Ut laoreet accumsan ex non porta. Donec vitae mi porttitor, dictum lacus sit amet, dapibus ipsum. Praesent bibendum lacus nibh, sed egestas enim facilisis vitae!
        </span>
      </span>
      <span
        className="pantheon-img-container pantheon-img-container-breakright"
        style={
          {
            "float": "right",
            "height": "auto",
            "marginTop": "170.85929748535153px",
            "maxHeight": "352.00px",
            "maxWidth": "480.00px",
            "shapeOutside": "padding-box",
            "width": "100%",
          }
        }
      >
        <img
          className=""
          src="https://cdn.example.com/loAWY0YB0HTHexSzw3Z1/djHVTYbPaCby44H5CxhH"
          style={
            {
              "height": "auto",
              "marginLeft": "0.00px",
              "maxHeight": "352.00px",
              "maxWidth": "480.00px",
              "width": "100%",
            }
          }
          title="just another dog"
        />
        <span
          className="pantheon-caption"
          style={
            {
              "display": "flex",
              "fontSize": ".75rem",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          just another dog
        </span>
      </span>
    </p>
  </div>
  <div>
    <style
      dangerouslySetInnerHTML={
        {
          "__html": "

ul.nesting-level-1 {
  list-style-type: disc; 
}
ul.nesting-level-2 {
  list-style-type: circle; 
}
ul.nesting-level-3 {
  list-style-type: square; 
}
ul.nesting-level-4 {
  list-style-type: disc; 
}
ul.nesting-level-5 {
  list-style-type: circle; 
}
ul.nesting-level-6 {
  list-style-type: square; 
}
ul.nesting-level-7 {
  list-style-type: disc; 
}
ul.nesting-level-8 {
  list-style-type: circle; 
}


ol[list-style-type="decimal"] {
  list-style-type: decimal;
}
ol[list-style-type="lower-alpha"] {
  list-style-type: lower-alpha;
}
ol[list-style-type="lower-roman"] { 
  list-style-type: lower-roman;
}

",
        }
      }
    />
    <p
      className="normal-text"
      style={
        {
          "margin": "0",
          "marginBottom": "11PT",
        }
      }
    >
      <span
        className=""
      >
        <span
          className=""
        >
          Integer pretium bibendum ipsum, vehicula placerat orci finibus id. Proin orci mauris, ornare commodo congue sit amet, dignissim non eros. Nullam risus nibh, dictum et enim vel, vestibulum maximus velit. Proin eu porta risus. Aliquam nec lacus et lacus commodo elementum. Mauris quis massa ac urna iaculis hendrerit. Nulla eget ullamcorper lorem, in rutrum sapien. Vivamus finibus placerat massa a mollis.
        </span>
      </span>
    </p>
    <p
      className="normal-text"
      style={
        {
          "margin": "0",
          "marginBottom": "11PT",
        }
      }
    >
      <span
        className=""
      >
        <span
          className=""
        >
          Cras vitae ultricies odio. Vestibulum iaculis interdum feugiat. Donec iaculis finibus lorem. Quisque lacinia volutpat magna, in dapibus lectus gravida id. Cras iaculis justo et diam elementum, nec rhoncus dolor efficitur. Sed lacus libero, lobortis in mauris ut, imperdiet suscipit quam. Sed eu leo vitae sapien vestibulum viverra.
        </span>
      </span>
    </p>
    <p
      className="normal-text"
      style={
        {
          "margin": "0",
          "marginBottom": "11PT",
        }
      }
    >
      <span
        className=""
      >
        <span
          className=""
        >
          Morbi varius, velit volutpat auctor malesuada, nisi mi rhoncus urna, at tempus risus libero ac turpis. Nullam sit amet porta leo. Nam dapibus maximus ante, vitae fringilla ipsum lacinia posuere. Donec efficitur, odio ac blandit convallis, purus arcu posuere risus, maximus scelerisque quam augue id ante. Etiam ultrices neque in purus rhoncus laoreet. Nunc bibendum tincidunt felis, quis molestie libero dictum scelerisque. Sed urna dolor, elementum eget auctor tincidunt, lacinia in diam. Duis a libero tristique, luctus lacus aliquet, fermentum odio. Integer lacinia nunc metus, ac egestas nibh mattis a. Fusce sodales mollis odio non pulvinar. Nulla vitae dignissim nibh, ac finibus neque.
        </span>
      </span>
    </p>
    <p
      className="normal-text"
      style={
        {
          "margin": "0",
          "marginBottom": "11PT",
        }
      }
    >
      <span
        className=""
      >
        <span
          className=""
        >
          Mauris mollis, lacus eu sodales tincidunt, ipsum ex pharetra libero, eu posuere ante lectus nec velit. Sed commodo ultrices diam eget pulvinar. Vestibulum volutpat sapien non nisl porttitor, sed euismod magna sollicitudin. Suspendisse potenti. Sed posuere sapien id magna euismod tincidunt. Phasellus in cursus diam, eget vestibulum ipsum. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Curabitur tempus mollis tempus. Phasellus rhoncus, turpis sed mollis rutrum, tellus mauris hendrerit nulla, a congue dolor mauris in leo. Sed feugiat posuere neque, eget cursus est luctus at. Vestibulum accumsan convallis ante in auctor. Aliquam ac orci sem. Nunc auctor mauris sed felis tristique, nec malesuada ante porta.
        </span>
      </span>
    </p>
    <p
      className="normal-text"
      style={
        {
          "margin": "0",
          "marginBottom": "11PT",
        }
      }
    >
      <span
        className=""
      >
        <span
          className=""
        >
          Sed et ligula nec nisl convallis volutpat. Curabitur semper massa ac mollis feugiat. Vestibulum eget orci ac arcu tempus dapibus. Nunc augue lorem, ultrices in velit non, porta cursus ex. Aenean eget nunc sit amet quam vulputate porttitor vitae sed ex. Morbi porttitor libero nec hendrerit vehicula. Ut maximus odio ac odio blandit pharetra. Etiam quis condimentum est. Morbi dui nisi, ornare at dignissim vel, ultrices rhoncus nisl. Sed at euismod enim. Quisque efficitur orci a velit pretium scelerisque. Nulla eleifend feugiat porta. Etiam eu rhoncus lectus.
        </span>
      </span>
      <span
        className="pantheon-img-container pantheon-img-container-breakleft"
        style={
          {
            "float": "left",
            "height": "auto",
            "marginTop": "24.937500000000085px",
            "maxHeight": "350.00px",
            "maxWidth": "470.00px",
            "shapeOutside": "padding-box",
            "width": "100%",
          }
        }
      >
        <img
          className=""
          src="https://cdn.example.com/loAWY0YB0HTHexSzw3Z1/ZWJn6ilXOYs9oS9iDgMH"
          style={
            {
              "height": "auto",
              "marginLeft": "0.00px",
              "maxHeight": "350.00px",
              "maxWidth": "470.00px",
              "width": "100%",
            }
          }
          title="This is a title"
        />
        <span
          className="pantheon-caption"
          style={
            {
              "display": "flex",
              "fontSize": ".75rem",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          This is a title
        </span>
      </span>
    </p>
    <p
      className="normal-text"
      style={
        {
          "margin": "0",
          "marginBottom": "11PT",
        }
      }
    >
      <span
        className=""
      >
        <span
          className=""
        >
          THIS PUPPY BREAKS UP TEXT Etiam maximus mauris eu felis interdum, in bibendum orci sodales. In posuere elit quam. Nulla dictum, sapien ut sagittis consequat, nisl elit suscipit lectus, a blandit risus nunc eget nunc. Donec augue ante, eleifend non risus at, dignissim gravida ipsum. Vivamus tristique tempor dui, at condimentum dui tristique nec. Suspendisse potenti. Nulla facilisi. Quisque orci risus, iaculis vel vehicula eu, cursus sit amet justo. Cras sed erat a est tristique elementum.
        </span>
      </span>
    </p>
    <p
      className="normal-text"
      style={
        {
          "margin": "0",
          "marginBottom": "11PT",
        }
      }
    >
      <span
        className=""
      >
        <span
          className=""
        >
          Morbi justo lorem, ultricies ut volutpat aliquam, blandit eu erat. Duis accumsan egestas convallis. Morbi a finibus lectus. Vivamus vel ultrices felis. Donec faucibus placerat mi, non pellentesque est lacinia non. Ut vitae metus leo. Mauris volutpat malesuada lorem, et auctor metus sodales ultrices. Nulla rhoncus augue pulvinar consectetur bibendum. Nunc ac urna vel mauris venenatis euismod in et massa.
        </span>
      </span>
    </p>
    <p
      className="normal-text"
      style={
        {
          "margin": "0",
          "marginBottom": "11PT",
        }
      }
    >
      <span
        className=""
      >
        <span
          className=""
        >
          Proin sed laoreet augue. Aliquam malesuada vel ex ut vestibulum. Mauris at libero vestibulum, vehicula mi nec, ullamcorper magna. Vivamus bibendum ipsum imperdiet placerat pharetra. Curabitur fermentum id est a tristique. Integer at semper erat. Phasellus volutpat dolor in ipsum hendrerit, eu maximus libero mattis. Nunc aliquam, tortor at congue vehicula, mi purus mollis metus, et convallis orci quam at elit.
        </span>
      </span>
      <span
        className="pantheon-img-container pantheon-img-container-breakright"
        style={
          {
            "float": "right",
            "height": "auto",
            "marginTop": "64.63423339843754px",
            "maxHeight": "352.00px",
            "maxWidth": "462.00px",
            "shapeOutside": "padding-box",
            "width": "100%",
          }
        }
      >
        <img
          className=""
          src="https://cdn.example.com/loAWY0YB0HTHexSzw3Z1/RoJOqZdOWPaxWCVnH9sD"
          style={
            {
              "height": "auto",
              "marginLeft": "0.00px",
              "maxHeight": "352.00px",
              "maxWidth": "462.00px",
              "width": "100%",
            }
          }
        />
      </span>
    </p>
    <p
      className="normal-text"
      style={
        {
          "margin": "0",
          "marginBottom": "11PT",
        }
      }
    >
      <span
        className=""
      >
        <span
          className=""
        >
          Etiam non eleifend nisl. Morbi eu libero a erat volutpat dictum eu sodales urna. Ut venenatis rutrum dolor, a euismod nibh egestas a. Nullam pretium congue tellus, eu tempor est consectetur et. Etiam dapibus consequat volutpat. Nunc sed bibendum ipsum, a egestas risus. Nulla id dolor nibh. Nulla tempor diam eget sapien elementum faucibus. Sed elementum, mi ut maximus facilisis, ex enim fringilla sem, at fermentum odio mi id diam. Pellentesque vitae commodo felis. Vestibulum eget diam pellentesque, porttitor ante a, ultricies lacus.
        </span>
      </span>
      <span
        className="pantheon-img-container pantheon-img-container-inline pantheon-img-container-alone"
        style={
          {
            "border": "0.00px solid #000000",
            "display": "inline-block",
            "height": "auto",
            "margin": "0.00px 0.00px",
            "maxHeight": "193.99px",
            "maxWidth": "256.50px",
            "overflow": "hidden",
            "width": "100%",
          }
        }
      >
        <img
          className=""
          src="https://cdn.example.com/loAWY0YB0HTHexSzw3Z1/RoJOqZdOWPaxWCVnH9sD"
          style={
            {
              "height": "auto",
              "marginLeft": "0.00px",
              "maxHeight": "193.99px",
              "maxWidth": "256.50px",
              "width": "100%",
            }
          }
        />
      </span>
    </p>
    <p
      className="normal-text"
      style={
        {
          "margin": "0",
        }
      }
    >
      <span
        className="pantheon-img-container pantheon-img-container-inline pantheon-img-container-alone"
        style={
          {
            "height": "auto",
            "maxHeight": "90.22325805664047px",
            "maxWidth": "118.41802619934062px",
            "width": "100%",
          }
        }
      >
        <img
          className=""
          src="https://cdn.example.com/loAWY0YB0HTHexSzw3Z1/RoJOqZdOWPaxWCVnH9sD"
          style={
            {
              "height": "auto",
              "marginLeft": "0.00px",
              "maxHeight": "90.45px",
              "maxWidth": "118.71px",
              "width": "100%",
            }
          }
        />
      </span>
    </p>
    <p
      className="normal-text"
      style={
        {
          "margin": "0",
        }
      }
    >
      <span
        className="pantheon-img-container pantheon-img-container-inline pantheon-img-container-alone"
        style={
          {
            "height": "auto",
            "maxHeight": "57.30575805664048px",
            "maxWidth": "75.77094676378019px",
            "width": "100%",
          }
        }
      >
        <img
          className=""
          src="https://cdn.example.com/loAWY0YB0HTHexSzw3Z1/RoJOqZdOWPaxWCVnH9sD"
          style={
            {
              "height": "auto",
              "marginLeft": "0.00px",
              "maxHeight": "57.45px",
              "maxWidth": "75.96px",
              "width": "100%",
            }
          }
          title="inline puppies"
        />
        <span
          className="pantheon-caption"
          style={
            {
              "display": "flex",
              "fontSize": ".75rem",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          inline puppies
        </span>
      </span>
    </p>
  </div>
</div>
`;
